pkg/devfile/parse.go

```diff
@@ -109,6 +109,15 @@ func ParseDevfileAndValidate(args parser.ParserArgs) (d parser.DevfileObj, varWa
 		varWarning = variables.ValidateAndReplaceGlobalVariable(d.Data.GetDevfileWorkspaceSpec())
 	}
 
+	// Use image names as selectors after variable substitution,
+	// as users might be using variables for image names.
+	if args.ImageNamesAsSelector != nil && args.ImageNamesAsSelector.Registry != "" {
```

Comment:
is `registry` a mandatory field? if it is, should mark as `required ` in the struct?

Reply:
Whoops, I missed your comment.

> is registry a mandatory field?
>
Initially, I didn't want to make it required. But with your comment, I think it does not make sense to pass a non-nil `ImageNamesAsSelector` with no registry inside.

> if it is, should mark as required  in the struct?
>
[`164ec3d` (#172)](https://github.com/devfile/library/pull/172/commits/164ec3d27a4041ede27bf7690d74a31f5459711c) makes it mandatory and checks if it is set when `ImageNamesAsSelector` is non-nil.